src/main/java/com/bt/code/egress/write/FolderWriter.java

```diff
@@ -49,6 +54,32 @@ public void onFileCompleted(FileCompleted fileCompleted) {
         }
     }
 
+    @Override
+    public void onZipCompleted(ZipCompleted zipCompleted) {
+        //Copy the resulting zip back in-place from temp dir
+        Path newZipAbsolutePath = getTempRoot().resolve(zipCompleted.getSourceZipRelativePath());
+
+        if (!Files.exists(newZipAbsolutePath)) {
+            //No changes to this zip file - skipping
+            return;
+        }
+
+        try {
+            Files.move(
```

Comment:
Would be good to log temp filec createion and removal for better tracking

Reply:
Added